xbmc/cores/dvdplayer/DVDPlayer.cpp

```diff
@@ -2593,7 +2593,8 @@ void CDVDPlayer::HandleMessages()
         if(input)
         {
           bool bSwitchSuccessful(false);
-          bool bShowPreview(CSettings::Get().GetInt("pvrplayback.channelentrytimeout") > 0);
+          bool bShowPreview(!g_infoManager.IsPlayerOSDActive() &&
+                            (CSettings::Get().GetInt("pvrplayback.channelentrytimeout") > 0));
```

Comment:
this won't work if you've not set the channel entry timeout but the confirm channel switch option.

This should be:
```
bool bShowPreview(!g_infoManager.IsPlayerOSDActive() &&
                            (CSettings::Get().GetBool("pvrplayback.confirmchannelswitch") ||
                             CSettings::Get().GetInt("pvrplayback.channelentrytimeout") > 0));
```